tests/e2e/global-setup.js

```diff
@@ -0,0 +1,81 @@
+const { chromium, expect } = require( '@playwright/test' );
+const { admin } = require( './config/default.json' ).users;
+const fs = require( 'fs' );
+
+/* eslint-disable no-console */
+module.exports = async ( config ) => {
+	const { stateDir, baseURL, userAgent } = config.projects[ 0 ].use;
+
+	console.log( `State Dir: ${ stateDir }` );
+	console.log( `Base URL: ${ baseURL }` );
+
+	// used throughout tests for authentication
+	process.env.ADMINSTATE = `${ stateDir }adminState.json`;
+
+	// Clear out the previous save states
+	try {
+		fs.unlinkSync( process.env.ADMINSTATE );
+		console.log( 'Admin state file deleted successfully.' );
+	} catch ( err ) {
+		if ( err.code === 'ENOENT' ) {
+			console.log( 'Admin state file does not exist.' );
+		} else {
+			console.log( 'Admin state file could not be deleted: ' + err );
+		}
+	}
+
+	// Pre-requisites
+	let adminLoggedIn = false;
+
+	// Specify user agent when running against an external test site to avoid getting HTTP 406 NOT ACCEPTABLE errors.
+	const contextOptions = { baseURL, userAgent };
+
+	// Create browser, browserContext, and page for customer and admin users
+	const browser = await chromium.launch();
+	const adminContext = await browser.newContext( contextOptions );
+	const adminPage = await adminContext.newPage();
+
+	// Sign in as admin user and save state
+	const adminRetries = 5;
+	for ( let i = 0; i < adminRetries; i++ ) {
+		try {
+			console.log( 'Trying to log-in as admin...' );
+			await adminPage.goto( `/wp-admin` );
+			await adminPage
+				.locator( 'input[name="log"]' )
+				.fill( admin.username );
+			await adminPage
+				.locator( 'input[name="pwd"]' )
+				.fill( admin.password );
+			await adminPage.locator( 'text=Log In' ).click();
+			await adminPage.waitForLoadState( 'networkidle' );
+			await adminPage.goto( `/wp-admin` );
+			await adminPage.waitForLoadState( 'domcontentloaded' );
+
+			await expect( adminPage.locator( 'div.wrap > h1' ) ).toHaveText(
+				'Dashboard'
+			);
+			await adminPage
+				.context()
+				.storageState( { path: process.env.ADMINSTATE } );
+			console.log( 'Logged-in as admin successfully.' );
+			adminLoggedIn = true;
+			break;
+		} catch ( e ) {
+			console.log(
+				`Admin log-in failed, Retrying... ${ i }/${ adminRetries }`
+			);
+			console.log( e );
```

Comment:
Maybe we can use `console.warn.` Also, I don't understand well why we need to retry 5 times?

Reply:
These are the parts that we need to do the same as WC core, but it isn't reusable to the same extent it was when we had the e2e packages. So it's basically copied code from [here](https://github.com/woocommerce/woocommerce/blob/7.9.0/plugins/woocommerce/tests/e2e-pw/global-setup.js#L53) with the comment further down:

> This step was failing occasionally, and globalsetup doesn't retry, so make it retry

I've had it fail in one of the test runs I did in GitHub Actions, so it must be a bit flakey at times (which is resolved with the retry).

Maybe I should add some references where the code is copied from to help explain. I can change the to console.warn / error where appropriate though.